apps/contracts/contracts/Zengo.sol

```diff
@@ -15,13 +15,33 @@ contract ZengoDAO is PermissionsEnumerable, ContractMetadata {
     address public votingTokenAddress; // Address of the ERC20 voting token
     address public deployer;
 
+    // Modify the Design to facilitate current implementation
     struct Proposal {
-        uint256 id;
+        uint256 proposalId;
         string title;
-        string description;
-        address submitter;
+        string proposalDescription;
+        string proposalType;
+        address proposer;
+        Evidence proposerEvidence;
+        Vote[] votingIterations;
+        // requires funding
+        bool isEligibleForFunding;
         bool isVerified;
-        uint256 votes;
+    }
+
+    struct Vote {
+        uint256 proposalId;
+        uint256 totalVotes;
+        mapping (address => bool) vote; 
```

Comment:
this mapping is handling the votes, right?
having something like:

```
// other fields
uint256[] votes;
uint256[] votesWeight;
string[] options;
```
where the votes are the votes count on eachoption 
and the votesWeight is the weight the vote represented to that option.
and with this data apply an `sqrt` math method to calculate winner/s ?

just asking :grinning:

Reply:
this vote has nothing to do with quadratic voting...this is only for the verification stage